modules/core/src/main/scala/ru/m2/calypso/Encoder.scala

```diff
@@ -3,94 +3,67 @@ package ru.m2.calypso
 import cats.data.NonEmptyList
 import org.bson.BsonValue
 import ru.m2.calypso.boilerplate.ProductEncoders
-import ru.m2.calypso.syntax._
+import ru.m2.calypso.syntax.*
 
 import java.time.Instant
 import java.util.UUID
 import scala.collection.immutable.{SortedMap, SortedSet}
 
-trait Encoder[A] {
+trait Encoder[A]:
   def apply(a: A): BsonValue
 
   final def contramap[B](f: B => A): Encoder[B] =
-    Encoder.instance { b =>
-      apply(f(b))
-    }
-}
+    b => apply(f(b))
 
-object Encoder extends ProductEncoders {
-  def apply[A](implicit instance: Encoder[A]): Encoder[A] = instance
+object Encoder extends ProductEncoders:
+  def apply[A](using encoder: Encoder[A]): Encoder[A] = encoder
 
   def instance[A](f: A => BsonValue): Encoder[A] = f(_)
 
-  implicit val encodeBsonValue: Encoder[BsonValue] = Encoder.instance(identity)
-
-  implicit val encodeBoolean: Encoder[Boolean] = Encoder.instance(Bson.boolean)
-
-  implicit val encodeInt: Encoder[Int] = Encoder.instance(Bson.int32)
-
-  implicit val encodeLong: Encoder[Long] = Encoder.instance(Bson.int64)
-
-  implicit val encodeDouble: Encoder[Double] = Encoder.instance(Bson.double)
-
-  implicit val encodeString: Encoder[String] = Encoder.instance(Bson.string)
-
-  implicit val encodeArrayByte: Encoder[Array[Byte]] = Encoder.instance(Bson.binary)
-
-  implicit val encodeUnit: Encoder[Unit] = Encoder.instance(_ => Bson.empty)
+  final def forCoproduct[A](f: A => (String, BsonValue)): Encoder[A] =
```

Comment:
methods and values within objects cannot be overridden, so `final` modifier is obsolete.